packages/dev/test/arbitrum/ArbRelayHub.test.ts

```diff
@@ -0,0 +1,118 @@
+import BN from 'bn.js'
+import { ether, expectEvent } from '@openzeppelin/test-helpers'
+
+import {
+  ArbRelayHubInstance,
+  ForwarderInstance,
+  StakeManagerInstance
+} from '@opengsn/contracts/types/truffle-contracts'
+import { constants, defaultEnvironment, environments, getEip712Signature } from '@opengsn/common'
+import { RelayRequest } from '@opengsn/common/dist/EIP712/RelayRequest'
+import { TypedRequestData } from '@opengsn/common/dist/EIP712/TypedRequestData'
+import { registerForwarderForGsn } from '@opengsn/common/dist/EIP712/ForwarderUtil'
+import { TransactionRelayed } from '@opengsn/contracts/types/truffle-contracts/RelayHub'
+
+const Forwarder = artifacts.require('Forwarder')
+const TestArbSys = artifacts.require('TestArbSys')
+const ArbRelayHub = artifacts.require('ArbRelayHub')
+const StakeManager = artifacts.require('StakeManager')
+const TestRecipient = artifacts.require('TestRecipient')
+const TestPaymasterEverythingAccepted = artifacts.require('TestPaymasterEverythingAccepted')
+
+contract.only('ArbRelayHub', function ([from, relayWorker, relayManager, relayOwner]: string[]) {
+  let arbRelayHub: ArbRelayHubInstance
+  let forwarder: ForwarderInstance
+  let stakeManager: StakeManagerInstance
+
+  before(async function () {
+    forwarder = await Forwarder.new()
+    stakeManager = await StakeManager.new(Number.MAX_SAFE_INTEGER)
+    const testArbSys = await TestArbSys.new()
+    arbRelayHub = await ArbRelayHub.new(testArbSys.address, stakeManager.address, constants.ZERO_ADDRESS, environments.arbitrum.relayHubConfiguration)
+  })
+
+  context('#aggregateGasleft()', function () {
+    it('should return gas left both for execution and for L2 storage', async function () {
```

Comment:
I don't understand why this test is needed

Reply:
this test indeed does nothing.
just like you can't really test "gasleft()" itself this way
you need to test it across some code (before/after), where we have some assumption on how much gas it should cost.
it has to be tested on arb node over a test that actually make some storage access - like the one we ran on actual arbitrum testnet.

this test is useless on hardhat.

Reply:
yes, not running arbitrum locally means I cannot "really" test anything, so I just write the tests that do nothing hoping they add a way to run Arbitrum locally and we add it to our test suite

Reply:
ok. didn't look into you TestArbSys before. I guess it is the best as it can get for a test on hardhat...